### ISSUE TITLE
Integer incorrectly narrowed to IntEnum

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#17860]
The revealed narrowed type is unexpected, and there is a false negative:
```py
from enum import IntEnum

class E(IntEnum):
    X = 1

def f(x: int) -> None:
    if x == E.X:
        print(x.value)  # No mypy error, but fails at runtime
        reveal_type(x)  # "Literal[E.X]"

f(1)
```
I'd expect the revealed type to be `int`, since that's the correct runtime type. IntEnum values are equal to the corresponding `int` objects:
```py
print(1 == E.X)  # True
```

Also, `x.value` should be flagged as an error, since it fails at runtime.

If this needs special casing for IntEnums, it's fine. We already special case enums in various ways.
